### PR TITLE
Adding P4RT App Configuration to HLD.

### PR DESCRIPTION
**Pull request by ntoorchi**
This HLD PR is corresponding to code PR #10725 against sonic-buildimage.

### CODE REVIEW

**Comment by ntoorchi** [comment]
@bocon13 please review.

File: doc/pins/p4rt_app_hld.md [hunk 1]
```diff
@@ -164,6 +165,30 @@ P4RT application introduces new tables that are written to APPL_DB for the table
 
 The P4RT application code will have unit & component tests that together will give >80% code coverage.
 
+## Configuring P4RT Application
+
+The P4RT application is configured at the start-up by reading the P4RT configuration from the CONFIG_DB. If no valid config exists in CONFIG_DB, it uses the default values. The configuration can be added to the CONFIG_DB either manually (by adding to config_db.json) or through gNMI, as it can be different for users.
```
Comment (rhalstea) [inline doc/pins/p4rt_app_hld.md]: I don't think most of these can be configured over gNMI. I would suggest leaving that part out.

Comment (rhalstea) [inline doc/pins/p4rt_app_hld.md]: Maybe also mention that the container will need to be restarted if the values are changed?

**Review by rhalstea — COMMENTED** [review]
Thanks Niloofar!

File: doc/pins/p4rt_app_hld.md [hunk 2]
```diff
@@ -164,6 +165,30 @@ P4RT application introduces new tables that are written to APPL_DB for the table
 
 The P4RT application code will have unit & component tests that together will give >80% code coverage.
 
+## Configuring P4RT Application
+
+The P4RT application is configured at the start-up by reading the P4RT configuration from the CONFIG_DB. If no valid config exists in CONFIG_DB, it uses the default values. The configuration can be added to the CONFIG_DB by being manually added to the config_db.json file. The P4RT container will need to be restarted if the configuration is changed.
+
+Below is an example of adding P4RT configuration to config_db.json. It initializes the P4RT application with the default configuration (which can be omitted). The user can modify this block based on their environment settings.
```
Comment (rhalstea) [inline doc/pins/p4rt_app_hld.md]: nit: I would remove this sentence:
```
It initializes the P4RT application with the default configuration (which can be omitted).
```
It's a bit redundant with the preceding paragraph.

File: doc/pins/p4rt_app_hld.md [hunk 3]
```diff
@@ -164,6 +165,30 @@ P4RT application introduces new tables that are written to APPL_DB for the table
...
+    "server_key": "/keys/server_key.lnk",
+    "ca_crt": "/keys/ca_cert.lnk",
+    "cert_crl_dir": "/keys/crl"
+  },
+  "p4rt_app": {
+    "port": "9559",
+    "use_genetlink": "false",
+    "use_port_ids": "false",
+    "save_forwarding_config_file" : "/etc/sonic/p4rt_forwarding_config.pb.txt",
+    "p4rt_unix_socket" : "/sock/p4rt.sock",
```
Comment (rhalstea) [inline doc/pins/p4rt_app_hld.md]: nit: I would also remove the `p4rt_unix_socket` argument.

While it is configurable, it has a very specific use case that I don't think most people staring in PINS would be interested in.

**Review by rhalstea — APPROVED** [review]
Thanks Niloofar, a few more nits, but looks good

**Comment by rhalstea** [comment]
@ntoorchi this looks good and the change has already gone in. Can we open a pull request against the main repo and ask Prince to pull in the changes?

**Pull request closed without merging**